src/index.js

```diff
@@ -1,32 +1,40 @@
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+//Usé el browser para correr y corregir este script. Razón por la cual el require no está presente.
 
 var API = 'https://rickandmortyapi.com/api/character/';
-var xhttp = new XMLHttpRequest();
-
-function fetchData(url_api, callback) {
-  xhttp.onreadystatechange = function (event) {
-    if (xhttp.readyState === '4') {
-      if (xhttp.status == 200)
-        callback(null, xhttp.responseText);
-      else return callback(url_api);
-    }
-  };
-  xhttp.open('GET', url_api, false);
-  xhttp.send();
+
+const fetchData = url_api => {
+	return new Promise((resolve, reject) => {
+		var xhttp = new XMLHttpRequest();
+		xhttp.responseType = 'json';
+
+		xhttp.onreadystatechange = function(event) {
+			if (xhttp.readyState === 4) {
+				if (xhttp.status == 200) resolve(xhttp.response);
```

Comment:
Te recomiendo ser mas claros en el reject, esto te permite identificar el error mas fácil. 

```suggestion
          reject(new Error("Algo salió mal."));
```